diff --git a/imagefield_crop.module b/imagefield_crop.module
index bc574af..4487dfc 100644
--- a/imagefield_crop.module
+++ b/imagefield_crop.module
@@ -403,7 +403,7 @@ function _imagefield_crop_entity_presave($entity, $fields) {
     if ($field['widget']['type'] != 'imagefield_crop_widget') {
       continue;
     }
-    if (!isset($entity->{$field_name})) {
+    if (empty($entity->{$field_name})) {
       continue;
     }
 
