diff --git a/paragraphs.module b/paragraphs.module
index 47093b6..abff860 100644
--- a/paragraphs.module
+++ b/paragraphs.module
@@ -1203,7 +1203,7 @@ function theme_paragraphs_field_multiple_value_form($variables) {
       );
       $rows[] = array(
         'data' => $cells,
-        'class' => array('draggable', drupal_html_class('paragraphs_item_type_' . $item['#bundle'])),
+        'class' => !empty($item['#bundle']) ? array('draggable', drupal_html_class('paragraphs_item_type_' . $item['#bundle'])) : '',
       );
     }
 
